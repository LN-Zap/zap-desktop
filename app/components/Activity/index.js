import Activity from './Activity'

export default Activity
