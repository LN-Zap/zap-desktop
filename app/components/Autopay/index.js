import Autopay from './Autopay'

export default Autopay
export AutopayActions from './AutopayActions'
export AutopayCardView from './AutopayCardView'
export AutopayHeader from './AutopayHeader'
export AutopayHeading from './AutopayHeading'
export AutopayList from './AutopayList'
export AutopaySearch from './AutopaySearch'
export AutopaySearchNoResults from './AutopaySearchNoResults'
