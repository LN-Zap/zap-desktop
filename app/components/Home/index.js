export Home from './Home'
export CreateWalletButton from './CreateWalletButton'
export NoWallets from './NoWallets'
export WalletsMenu from './WalletsMenu'
export WalletLauncher from './WalletLauncher'
export WalletUnlocker from './WalletUnlocker'
export WalletSettingsFormLocal from './WalletSettingsFormLocal'
export WalletSettingsFormRemote from './WalletSettingsFormRemote'
export WalletHeader from './WalletHeader'
