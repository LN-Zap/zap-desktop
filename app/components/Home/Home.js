import React from 'react'
import PropTypes from 'prop-types'
import { Route, Switch, withRouter } from 'react-router-dom'
import { Box } from 'rebass'
import { Bar, Heading, MainContent, Sidebar } from 'components/UI'
import ZapLogo from 'components/Icon/ZapLogo'
import { CreateWalletButton, WalletLauncher, WalletsMenu, WalletUnlocker } from '.'

const NoMatch = () => (
  <Box>
    <Heading>Please select a wallet</Heading>
  </Box>
)

class Home extends React.Component {
  static propTypes = {
    activeWallet: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    deleteWallet: PropTypes.func.isRequired,
    history: PropTypes.object.isRequired,
    lightningGrpcActive: PropTypes.bool.isRequired,
    walletUnlockerGrpcActive: PropTypes.bool.isRequired,
    wallets: PropTypes.array.isRequired,
    startLnd: PropTypes.func.isRequired,
    stopLnd: PropTypes.func.isRequired,
    setActiveWallet: PropTypes.func.isRequired,
    unlockWallet: PropTypes.func.isRequired,
    setUnlockWalletError: PropTypes.func.isRequired,
    unlockingWallet: PropTypes.bool,
    unlockWalletError: PropTypes.string
  }

  /**
   * Handle click event on the Create new wallet button,
   */
  handleCreateNewWalletClick = () => {
    const { history } = this.props
    history.push('/onboarding')
  }

  render() {
    const {
      activeWallet,
      deleteWallet,
      startLnd,
      unlockWallet,
      wallets,
      setActiveWallet,
      stopLnd,
      lightningGrpcActive,
      walletUnlockerGrpcActive,
      setUnlockWalletError,
      unlockingWallet,
      unlockWalletError
    } = this.props

    return (
      <>
        <Sidebar.small p={3} pt={40}>
          <ZapLogo width="70px" height="32px" />

          <WalletsMenu
            wallets={wallets}
            mt={30}
            activeWallet={activeWallet}
            setActiveWallet={setActiveWallet}
          />

          <Box width={1} css={{ position: 'absolute', left: 0, bottom: 0 }} px={3}>
            <Bar mx={-3} />
            <CreateWalletButton onClick={this.handleCreateNewWalletClick} width={1} p={3} />
          </Box>
        </Sidebar.small>

        <MainContent>
          <Box px={5} mt={72}>
            <Switch>
              <Route
                exact
                path="/home/wallet/:walletId"
                render={({ match: { params } }) => {
                  const wallet = wallets.find(wallet => wallet.id == params.walletId)
                  if (!wallet) {
                    return null
                  }
                  return (
                    <WalletLauncher
                      wallet={wallet}
                      startLnd={startLnd}
                      stopLnd={stopLnd}
                      lightningGrpcActive={lightningGrpcActive}
                      walletUnlockerGrpcActive={walletUnlockerGrpcActive}
                      deleteWallet={deleteWallet}
                    />
                  )
                }}
              />
              <Route
                exact
                path="/home/wallet/:walletId/unlock"
                render={({ match: { params } }) => {
                  const wallet = wallets.find(wallet => wallet.id == params.walletId)
                  if (!wallet) {
                    return null
                  }
                  return (
                    <WalletUnlocker
                      wallet={wallet}
                      unlockWallet={unlockWallet}
                      lightningGrpcActive={lightningGrpcActive}
                      setUnlockWalletError={setUnlockWalletError}
                      unlockingWallet={unlockingWallet}
                      unlockWalletError={unlockWalletError}
                    />
                  )
                }}
              />
              <Route component={NoMatch} />
            </Switch>
          </Box>
        </MainContent>
      </>
    )
  }
}

export default withRouter(Home)
