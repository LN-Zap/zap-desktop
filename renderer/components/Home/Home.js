import React from 'react'
import PropTypes from 'prop-types'
import { Redirect, Route, Switch, withRouter } from 'react-router-dom'
import { Box, Flex } from 'rebass/styled-components'
import { Bar, MainContent, Panel, Sidebar } from 'components/UI'
import ZapLogo from 'components/Icon/ZapLogo'
import CreateWalletButton from './CreateWalletButton'
import NoWallets from './NoWallets'
import WalletLauncher from './WalletLauncher'
import WalletsMenu from './WalletsMenu'
import WalletUnlocker from './WalletUnlocker'

const NoMatch = ({ history, wallets }) => (
  <Flex alignItems="center" flexDirection="column" height="100%" justifyContent="center">
    <NoWallets history={history} wallets={wallets} />
  </Flex>
)
NoMatch.propTypes = {
  history: PropTypes.object.isRequired,
  wallets: PropTypes.array.isRequired,
}

class Home extends React.Component {
  static propTypes = {
    activeWallet: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    activeWalletSettings: PropTypes.object,
    clearStartLndError: PropTypes.func.isRequired,
    deleteWallet: PropTypes.func.isRequired,
    generateLndConfigFromWallet: PropTypes.func.isRequired,
    history: PropTypes.object.isRequired,
    isLightningGrpcActive: PropTypes.bool.isRequired,
    isNeutrinoRunning: PropTypes.bool.isRequired,
    isStartingLnd: PropTypes.bool,
    isUnlockingWallet: PropTypes.bool,
    isWalletUnlockerGrpcActive: PropTypes.bool.isRequired,
    lndConnect: PropTypes.string,
    putWallet: PropTypes.func.isRequired,
    setActiveWallet: PropTypes.func.isRequired,
    setIsWalletOpen: PropTypes.func.isRequired,
    setUnlockWalletError: PropTypes.func.isRequired,
    showError: PropTypes.func.isRequired,
    showNotification: PropTypes.func.isRequired,
    startLnd: PropTypes.func.isRequired,
    startLndError: PropTypes.object,
    stopLnd: PropTypes.func.isRequired,
    unlockWallet: PropTypes.func.isRequired,
    unlockWalletError: PropTypes.string,
    wallets: PropTypes.array.isRequired,
  }

  // If there is an active wallet ensure it is selected on mount.
  componentDidMount() {
    const { activeWallet, activeWalletSettings, history, setIsWalletOpen } = this.props
    if (activeWallet && activeWalletSettings && history.location.pathname === '/home') {
      history.push(`/home/wallet/${activeWallet}`)
    }

    setIsWalletOpen(false)
  }

  componentDidUpdate(prevProps) {
    const { lndConnect, history } = this.props
    if (lndConnect && lndConnect !== prevProps.lndConnect) {
      history.push(`/onboarding`)
    }
  }

  render() {
    const {
      history,
      activeWallet,
      deleteWallet,
      startLnd,
      startLndError,
      isStartingLnd,
      unlockWallet,
      wallets,
      setActiveWallet,
      clearStartLndError,
      showError,
      stopLnd,
      isNeutrinoRunning,
      isLightningGrpcActive,
      isWalletUnlockerGrpcActive,
      setUnlockWalletError,
      isUnlockingWallet,
      unlockWalletError,
      putWallet,
      showNotification,
      generateLndConfigFromWallet,
    } = this.props

    return (
      <>
        <Sidebar.medium pt={40}>
          <Panel>
            <Panel.Header mb={40} px={4}>
              <ZapLogo height={28} width={28} />
            </Panel.Header>
            <Panel.Body sx={{ overflowY: 'overlay' }}>
              <WalletsMenu
                activeWallet={activeWallet}
                setActiveWallet={setActiveWallet}
                wallets={wallets}
              />
            </Panel.Body>
            <Panel.Footer>
              <Bar variant="light" />
              <Box px={4} py={2}>
                <CreateWalletButton history={history} width={1} />
              </Box>
            </Panel.Footer>
          </Panel>
        </Sidebar.medium>

        <MainContent pb={2} pl={5} pr={6} pt={4}>
          <Switch>
            <Route
              exact
              path="/home/wallet/:walletId"
              render={({ match: { params } }) => {
                const wallet = wallets.find(wallet => wallet.id == params.walletId)
                if (!wallet) {
                  return <Redirect to="/home" />
                }
                return (
                  <WalletLauncher
                    key={wallet.id}
                    clearStartLndError={clearStartLndError}
                    deleteWallet={deleteWallet}
                    generateLndConfigFromWallet={generateLndConfigFromWallet}
                    isLightningGrpcActive={isLightningGrpcActive}
                    isNeutrinoRunning={isNeutrinoRunning}
                    isStartingLnd={isStartingLnd}
                    isWalletUnlockerGrpcActive={isWalletUnlockerGrpcActive}
                    putWallet={putWallet}
                    showError={showError}
                    showNotification={showNotification}
                    startLnd={startLnd}
                    startLndError={startLndError}
                    stopLnd={stopLnd}
                    wallet={wallet}
                  />
                )
              }}
            />
            <Route
              exact
              path="/home/wallet/:walletId/unlock"
              render={({ match: { params } }) => {
                const wallet = wallets.find(wallet => wallet.id == params.walletId)
                if (!wallet) {
                  return <Redirect to="/home" />
                }
                return (
                  <WalletUnlocker
                    key={wallet.id}
                    isLightningGrpcActive={isLightningGrpcActive}
                    isUnlockingWallet={isUnlockingWallet}
                    setUnlockWalletError={setUnlockWalletError}
                    unlockWallet={unlockWallet}
                    unlockWalletError={unlockWalletError}
                    wallet={wallet}
                  />
                )
              }}
            />
            <Route render={() => <NoMatch history={history} wallets={wallets} />} />
          </Switch>
        </MainContent>
      </>
    )
  }
}

export default withRouter(Home)
