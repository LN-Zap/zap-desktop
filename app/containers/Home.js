import { connect } from 'react-redux'
import { setActiveWallet, walletSelectors, deleteWallet } from 'reducers/wallet'
import { setUnlockWalletError, stopLnd, startLnd, unlockWallet } from 'reducers/lnd'
import { Home } from 'components/Home'

const mapStateToProps = state => ({
  wallets: state.wallet.wallets,
  activeWallet: walletSelectors.activeWallet(state),
  activeWalletSettings: walletSelectors.activeWalletSettings(state),
  lightningGrpcActive: state.lnd.lightningGrpcActive,
  walletUnlockerGrpcActive: state.lnd.walletUnlockerGrpcActive,
  unlockingWallet: state.lnd.unlockingWallet,
  unlockWalletError: state.lnd.unlockWalletError
})

const mapDispatchToProps = {
  setActiveWallet,
  setUnlockWalletError,
  stopLnd,
  startLnd,
  unlockWallet,
  deleteWallet
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Home)
