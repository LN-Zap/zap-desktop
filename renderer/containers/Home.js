import React from 'react'
import { connect } from 'react-redux'
import {
  setActiveWallet,
  walletSelectors,
  showDeleteWalletDialog,
  setIsWalletOpen,
  putWallet,
} from 'reducers/wallet'
import {
  setUnlockWalletError,
  stopLnd,
  startLnd,
  unlockWallet,
  refreshLndConnectURI,
  clearStartLndError,
} from 'reducers/lnd'
import { showError, showNotification } from 'reducers/notification'
import Home from 'components/Home'
import DeleteWalletDialog from './Home/DeleteWalletDialog'

const HomeWrapper = props => (
  <>
    <Home {...props} />
    <DeleteWalletDialog />
  </>
)

const mapStateToProps = state => ({
  lndConnect: state.onboarding.lndConnect,
  wallets: state.wallet.wallets,
  activeWallet: walletSelectors.activeWallet(state),
  activeWalletSettings: walletSelectors.activeWalletSettings(state),
  isLightningGrpcActive: state.lnd.isLightningGrpcActive,
  isWalletUnlockerGrpcActive: state.lnd.isWalletUnlockerGrpcActive,
  startLndError: state.lnd.startLndError,
  isStartingLnd: state.lnd.isStartingLnd,
  isUnlockingWallet: state.lnd.isUnlockingWallet,
  unlockWalletError: state.lnd.unlockWalletError,
})

const mapDispatchToProps = {
  setActiveWallet,
  setUnlockWalletError,
  clearStartLndError,
  stopLnd,
  putWallet,
  showNotification,
  startLnd,
  unlockWallet,
  deleteWallet: showDeleteWalletDialog,
  setIsWalletOpen,
  showError,
  refreshLndConnectURI,
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(HomeWrapper)
