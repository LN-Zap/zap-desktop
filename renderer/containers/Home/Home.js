import React from 'react'
import { connect } from 'react-redux'
import { neutrinoSelectors } from 'reducers/neutrino'
import {
  setActiveWallet,
  walletSelectors,
  setIsWalletOpen,
  putWallet,
  DELETE_WALLET_DIALOG_ID,
} from 'reducers/wallet'
import {
  setUnlockWalletError,
  stopLnd,
  startLnd,
  unlockWallet,
  generateLndConfigFromWallet,
  clearStartLndError,
} from 'reducers/lnd'
import { openDialog } from 'reducers/modal'
import { showError, showNotification } from 'reducers/notification'
import Home from 'components/Home'
import DeleteWalletDialog from './DeleteWalletDialog'
import AppErrorBoundary from 'components/ErrorBoundary/AppErrorBoundary'

const deleteWallet = openDialog.bind(null, DELETE_WALLET_DIALOG_ID)

const HomeWrapper = props => (
  <>
    <Home {...props} />
    <DeleteWalletDialog />
  </>
)

const mapStateToProps = state => ({
  lndConnect: state.onboarding.lndConnect,
  wallets: state.wallet.wallets,
  activeWallet: walletSelectors.activeWallet(state),
  activeWalletSettings: walletSelectors.activeWalletSettings(state),
  isNeutrinoRunning: neutrinoSelectors.isNeutrinoRunning(state),
  isLightningGrpcActive: state.lnd.isLightningGrpcActive,
  isWalletUnlockerGrpcActive: state.lnd.isWalletUnlockerGrpcActive,
  startLndError: state.lnd.startLndError,
  isStartingLnd: state.lnd.isStartingLnd,
  isUnlockingWallet: state.lnd.isUnlockingWallet,
  unlockWalletError: state.lnd.unlockWalletError,
})

const mapDispatchToProps = {
  setActiveWallet,
  setUnlockWalletError,
  clearStartLndError,
  stopLnd,
  putWallet,
  showNotification,
  startLnd,
  unlockWallet,
  deleteWallet,
  setIsWalletOpen,
  showError,
  generateLndConfigFromWallet,
}

const ConnectedHome = connect(
  mapStateToProps,
  mapDispatchToProps
)(HomeWrapper)

const HomeWithErrorBoundaries = props => (
  <AppErrorBoundary>
    <ConnectedHome {...props} />
  </AppErrorBoundary>
)

export default HomeWithErrorBoundaries
