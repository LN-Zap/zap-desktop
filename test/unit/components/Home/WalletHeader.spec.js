import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { WalletHeader } from 'components/Home'

describe('component.WalletHeader', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<WalletHeader title="wallet 1" />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
