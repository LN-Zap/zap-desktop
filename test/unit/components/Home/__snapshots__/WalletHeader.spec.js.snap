// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.WalletHeader should render correctly 1`] = `
<Heading1>
  <TextFit
    autoResize={true}
    forceSingleModeWidth={true}
    max={60}
    min={16}
    mode="single"
    onReady={[Function]}
    throttle={50}
  >
    wallet 1
  </TextFit>
</Heading1>
`;
